src/main/java/net/snowflake/client/jdbc/SnowflakeFileTransferAgent.java

```diff
@@ -2319,18 +2317,15 @@ private void filterExistingFiles() throws SnowflakeSQLException {
                     ex.getCause(); // Cause of StorageProviderException is always an Exception
           }
           storageClient.handleStorageException(ex, ++retryCount, "listObjects", session, command);
+          continue;
         }
-      } while (retryCount <= storageClient.getMaxRetries());
 
-      logger.debug("received object summaries from remote storage");
-
-      retryCount = 0;
-      do {
         try {
           compareAndSkipRemoteFiles(objectSummaries, destFileNameToSrcFileMap);
-          break;
+          break; // exit retry loop
```

Comment:
Before, here are two do-while loops. One for `compareAndSkipRemoteFiles` and one for `storageClient.listObjects`. Now we merge `compareAndSkipRemoteFiles` into above `storageClient.listObjects`'s do-while loop, such that `objectSummaries` can be regenerated no matter exception happens in `storageClient.listObjects` or `compareAndSkipRemoteFiles`. Previously, if the exception happens in `compareAndSkipRemoteFiles`, the `objectSummaries` won't be regenerated, only `storageClient` will be refreshed.